crates/syntax/src/ast/make.rs

```diff
@@ -158,6 +158,49 @@ fn ty_from_text(text: &str) -> ast::Type {
     ast_from_text(&format!("type _T = {text};"))
 }
 
+/// Related goto [link](https://doc.rust-lang.org/reference/items/type-aliases.html)
+///     Type Alias syntax is
+///
+///     ```
+///     TypeAlias :
+///        type IDENTIFIER GenericParams? ( : TypeParamBounds )? WhereClause? ( = Type WhereClause?)? ;
+///     ```
+///
+///     FIXME : ident should be of type ast::Ident
+pub fn ty_alias(
+    ident: &str,
+    generic_param_list: Option<ast::GenericParamList>,
+    type_param_bounds: Option<ast::TypeParam>,
+    where_clause: Option<ast::WhereClause>,
+    assignment: Option<(ast::Type, Option<ast::WhereClause>)>,
+) -> ast::TypeAlias {
+    let mut s = String::new();
+    s.push_str(&format!("type {}", ident));
+
+    if let Some(list) = generic_param_list {
+        s.push_str(&list.to_string());
+    }
+
+    if let Some(list) = type_param_bounds {
+        s.push_str(&format!(" : {}", &list.to_string()));
```

Comment:
Ah actually, you can even drop the `to_string`! the ast nodes themselves implement `Display`